From: Benjamin Otte <otte@redhat.com>
Date: Fri, 8 Nov 2024 15:46:21 +0100
Subject: Revert "Fix the icon-effect-missing reftest"

This reverts commit 4f81dc1eafc13595841ab404dc8070909500dafa.

This change is no longer necessary after the last commit.

Part-of: <https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/8461>
(cherry picked from commit b8f6aea14eedacbdbd124a6f3a564c05b8fbde26)

Origin: upstream gtk-4-18 after 4.18.4
---
 testsuite/reftests/icon-effect-missing.ref.ui |  1 -
 testsuite/reftests/icon-theme-hicolor.c       | 35 ---------------------------
 testsuite/reftests/meson.build                |  1 -
 3 files changed, 37 deletions(-)
 delete mode 100644 testsuite/reftests/icon-theme-hicolor.c

diff --git a/testsuite/reftests/icon-effect-missing.ref.ui b/testsuite/reftests/icon-effect-missing.ref.ui
index ea916ba..827aaa9 100644
--- a/testsuite/reftests/icon-effect-missing.ref.ui
+++ b/testsuite/reftests/icon-effect-missing.ref.ui
@@ -2,7 +2,6 @@
 <interface>
   <object class="GtkWindow" id="window1">
     <property name="decorated">0</property>
-    <signal name="map" handler="reftest:set_icon_theme_hicolor" swapped="no"/>
     <child>
       <object class="GtkBox" id="box1">
         <child>
diff --git a/testsuite/reftests/icon-theme-hicolor.c b/testsuite/reftests/icon-theme-hicolor.c
deleted file mode 100644
index 3934fd8..0000000
--- a/testsuite/reftests/icon-theme-hicolor.c
+++ /dev/null
@@ -1,35 +0,0 @@
-/*
- * Copyright (C) 2021 Red Hat, Inc
- *
- * Author:
- *      Matthias Clasen <mclasen@redhat.com>
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Library General Public
- * License as published by the Free Software Foundation; either
- * version 2 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Library General Public License for more details.
- *
- * You should have received a copy of the GNU Library General Public
- * License along with this library. If not, see <http://www.gnu.org/licenses/>.
- */
-
-#include "config.h"
-
-#include <gtk/gtk.h>
-
-
-G_MODULE_EXPORT void
-set_icon_theme_hicolor (void)
-{
-  GtkSettings *settings;
-
-  g_test_message ("Attention: setting icon-theme to hicolor");
-
-  settings = gtk_settings_get_for_display (gdk_display_get_default ());
-  g_object_set (settings, "gtk-icon-theme-name", "hicolor", NULL);
-}
diff --git a/testsuite/reftests/meson.build b/testsuite/reftests/meson.build
index ab08da9..3cc92bd 100644
--- a/testsuite/reftests/meson.build
+++ b/testsuite/reftests/meson.build
@@ -25,7 +25,6 @@ libreftest = library('reftest',
     'textview-border-windows.c',
     'textview-tags.c',
     'animation-direction.c',
-    'icon-theme-hicolor.c',
   ],
   link_with: libgtkreftestprivate,
   c_args: reftest_cflags,
